diff -Naur zlib-1.2.8/contrib/minizip/unzip.c zlib-nocrc/contrib/minizip/unzip.c
--- zlib-1.2.8/contrib/minizip/unzip.c	2013-02-24 08:32:30.000000000 +0100
+++ zlib-nocrc/contrib/minizip/unzip.c	2014-11-20 02:20:12.048594880 +0100
@@ -1434,7 +1434,7 @@
 
     if (unz64local_getLong(&s->z_filefunc, s->filestream,&uData) != UNZ_OK) /* crc */
         err=UNZ_ERRNO;
-    else if ((err==UNZ_OK) && (uData!=s->cur_file_info.crc) && ((uFlags & 8)==0))
+    else if ((err==UNZ_OK) && ((uFlags & 8)==0))
         err=UNZ_BADZIPFILE;
 
     if (unz64local_getLong(&s->z_filefunc, s->filestream,&uData) != UNZ_OK) /* size compr */
